solr/webapp/web/js/angular/controllers/cloud.js

```diff
@@ -198,10 +194,13 @@ var nodesSubController = function($scope, Collections, System, Metrics) {
           for (var replicaName in replicas) {
             var core = replicas[replicaName];
             core.name = replicaName;
-            core.label = coreNameToLabel(core['core']);
+            core.replica = core['core'].replace(/.*_(replica_.*)$/, '\$1');
             core.collection = collection.name;
             core.shard = shard.name;
             core.shard_state = shard.state;
+            core.label = core['collection'] + "_"
+              + (core['shard'] + "_").replace(/shard(\d+)_/, 's\$1')
+              + core['replica'].replace(/replica_?[ntp]?(\d+)/, 'r\$1');
```

Comment:
I think could be made a bit clearer as to what is going here in the GitHub description, e.g. super imposing a box on what has changed  or a before/after, which is somewhat standard for UI. Alternatively, you could show how some of the names were treated in the past and how they will be treated going forward. Otherwise, mostly looks good.

However, I know this is in maintenance mode.

Reply:
My only question is, what is the ntp here for? I'm assuming NRT, Pull, and TLog replicas?

Reply:
Are there any reasons why we would not do any processing whatsoever of the shard names and simply show them as they were named? I assume there are some other screens or other confusion that could arise, but I am trying to doubly check that I understand the approach here fully.

Reply:
The label is for display purpose only. There is no before/after. This code produces exactly same display label as previously, but now also works for the custom shard names in implicit mode.

Reply:
Yes, `ntp` is replica type.

Reply:
Does the test failure have anything to do with the change? It doesn't seem like it.

Reply:
so, when implicit mode what would happen before this change?

After, it will show up. It would be helpful to know if (1) there was a particular error, (2) didn't show up at all, (3) was a broken page, etc.

Reply:
> so, when implicit mode what would happen before this change?

The label would be ok, but we’d use wrong key for fetching metrics and get empty values in UI.